sdk/storage/Azure.Storage.Blobs/src/AssemblyInfo.cs

```diff
@@ -2,43 +2,10 @@
 // Licensed under the MIT License.
 
 using System.Runtime.CompilerServices;
-
-#pragma warning disable AZC0011 // Avoid internals visible to product assemblies
-[assembly: InternalsVisibleTo("Azure.Storage.Blobs.Cryptography, PublicKey=" +
-    "0024000004800000940000000602000000240000525341310004000001000100097AD52ABBEAA2" +
-    "E1A1982747CC0106534F65CFEA6707EAED696A3A63DAEA80DE2512746801A7E47F88E7781E71AF" +
-    "960D89BA2E25561F70B0E2DBC93319E0AF1961A719CCF5A4D28709B2B57A5D29B7C09DC8D269A4" +
-    "90EBE2651C4B6E6738C27C5FB2C02469FE9757F0A3479AC310D6588A50A28D7DD431B907FD325E" +
-    "18B9E8ED")]
-[assembly: InternalsVisibleTo("Azure.Storage.Blobs.Batch, PublicKey=" +
-    "0024000004800000940000000602000000240000525341310004000001000100097AD52ABBEAA2" +
-    "E1A1982747CC0106534F65CFEA6707EAED696A3A63DAEA80DE2512746801A7E47F88E7781E71AF" +
-    "960D89BA2E25561F70B0E2DBC93319E0AF1961A719CCF5A4D28709B2B57A5D29B7C09DC8D269A4" +
-    "90EBE2651C4B6E6738C27C5FB2C02469FE9757F0A3479AC310D6588A50A28D7DD431B907FD325E" +
-    "18B9E8ED")]
-[assembly: InternalsVisibleTo("Azure.Storage.Files.DataLake, PublicKey=" +
-    "0024000004800000940000000602000000240000525341310004000001000100097AD52ABBEAA2" +
-    "E1A1982747CC0106534F65CFEA6707EAED696A3A63DAEA80DE2512746801A7E47F88E7781E71AF" +
-    "960D89BA2E25561F70B0E2DBC93319E0AF1961A719CCF5A4D28709B2B57A5D29B7C09DC8D269A4" +
-    "90EBE2651C4B6E6738C27C5FB2C02469FE9757F0A3479AC310D6588A50A28D7DD431B907FD325E" +
-    "18B9E8ED")]
-#pragma warning restore AZC0011
 [assembly: InternalsVisibleTo("Azure.Storage.Blobs.Tests, PublicKey=" +
```

Comment:
How hard is it to also remove the internals dependency from the test projects?

Reply:
Hard.

Reply:
To elaborate, since I'm being thumbed downward, we're constantly being told to have fewer public APIs whenever possible.  This results in a bunch of `internal` functionality that we'd still like to unit test without jumping through private reflection hoops.  We've reduced the internal visibility down to a direct mapping from class library to corresponding test project (i.e., Common to just Common.Tests, not Blobs.Tests) and I think it's reasonable.

If EngSys wants to setup a flag we can wrap these behind so they're not included in the public version when publishing, it won't be a breaking change and I'll happily adopt it.

Reply:
The thumbs down is more about how I hate tests having internals visible to and less about the statement that it might be hard to do. You can read more about my concerns in an older corefx issue https://github.com/dotnet/corefx/issues/1604. I personally believe it leads us to testing the wrong things because it is exercising code paths that the customer is not.